euphonic/spectra/collections.py

```diff
@@ -234,13 +235,16 @@ def _validate_item(self, item: Integral | slice | Sequence[Integral] | np.ndarra
             return
         if isinstance(item, slice):
             if (item.stop is not None) and (item.stop >= len(self)):
```

Comment:
There is a (little known) [`slice.indices`](https://docs.python.org/3/reference/datamodel.html#slice.indices) method for this.

Reply:
Created issue